### ISSUE TITLE
Host continously appears as "Registering" on Rancher server

### ISSUE THREAD

**Issue by mrleonard** [rancher/rancher#6227]
**Rancher Version:** 1.2.0-pre4-rc2

**Docker Version:** 1.12.1

**OS and where are the hosts located? (cloud, bare metal, etc):** bare metal, Ubuntu 16.04.1

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** single node with internal DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

**Steps to Reproduce:**
Install Rancher Server
Configure access control with Active Directory
Clic on Add a host button
Copy/paste the url on the host
use `docker logs -f rancher-agent` to see the rancher-agent container logs

**Results:** 
The host appears as "Registering" in the Infrastructure view on Rancher UI
the log are showing : 
`time="2016-10-11T08:27:32Z" level=info msg="Reply: fd1603e3-291c-45c5-55d8-5d31e6fe4bd0, reply.1227211772349627791, 1a1:agent, data: not empty" 
time="2016-10-11T08:27:37Z" level=info msg="Reply: 586a4a7c-4ba6-428f-4359-7fff4f54c97b, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:27:42Z" level=info msg="Reply: e50545a2-7c55-45b8-66d4-b959816dc3df, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:27:47Z" level=info msg="Reply: aadcc678-e55b-4b54-662f-1f1b8ce3837a, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:27:52Z" level=info msg="Reply: 73db38aa-8576-40c8-53ac-dc1c36d235de, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:27:57Z" level=info msg="Reply: 68026a3f-e77f-4da2-5ea2-77e5ca3e5e89, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:28:02Z" level=info msg="Reply: 30a03372-d4fb-4d8c-6798-644e886129e2, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:28:07Z" level=info msg="Reply: 715123e2-6b93-4454-6aeb-ee6abc33f2d2, reply.6657127758971742808, 1a1:agent, data: not empty" 
time="2016-10-11T08:28:12Z" level=info msg="Reply: 468de072-f7a9-4397-5493-d6b5203d857b, reply.6657127758971742808, 1a1:agent, data: not empty" `

**Expected:**
The host should be active

**Comment by janeczku** [comment]
Just happened here as well on 1.2.0-pre4-rc2 with host provisioned with DO integration.
rancher-server log:
```
2016-10-15 22:01:01,588 ERROR [9dbe4e45-69a8-49f8-89c2-4939556810f3:231] [host:3] [host.create] [] [erviceReplay-13] [c.p.e.p.i.DefaultProcessInstanceImpl] Unknown exception java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.String
	at io.cattle.platform.process.host.HostCreateUpdate.handle(HostCreateUpdate.java:65) ~[cattle-iaas-logic-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runHandler(DefaultProcessInstanceImpl.java:427) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:378) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:372) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.idempotent.Idempotent.execute(Idempotent.java:42) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runHandlers(DefaultProcessInstanceImpl.java:372) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runLogic(DefaultProcessInstanceImpl.java:474) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runWithProcessLock(DefaultProcessInstanceImpl.java:308) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$2.doWithLockNoResult(DefaultProcessInstanceImpl.java:247) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.acquireLockAndRun(DefaultProcessInstanceImpl.java:244) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runDelegateLoop(DefaultProcessInstanceImpl.java:186) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.executeWithProcessInstanceLock(DefaultProcessInstanceImpl.java:159) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:108) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:105) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.execute(DefaultProcessInstanceImpl.java:105) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.eventing.impl.ProcessEventListenerImpl.processExecute(ProcessEventListenerImpl.java:75) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.server.impl.ProcessInstanceParallelDispatcher$1.runInContext(ProcessInstanceParallelDispatcher.java:27) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) [na:1.7.0_111]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_111]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_111]

```

**Comment by deniseschannon** [comment]
There are known issues for v1.2.0-pre4-rc2 regarding host/machine (https://github.com/rancher/rancher/issues/6193) that have been fixed in rancher/server:master and should be fixed when v1.2.0-pre4-rc3. 

@mrleonard @janeczku If you still face this issue with either the latest versions, please re-open.
